zanata-common-api/src/main/java/org/zanata/common/DocumentType.java

```diff
@@ -41,37 +45,75 @@
 
     SUBTITLE("srt", "sbt", "sub", "vtt"),
-    PROPERTIES("properties"), XML("xml");
+    PROPERTIES("properties"), PROPERTIES_UTF8("properties"),
+    XML("xml"), XLIFF("xml");
+
+    private static final Set<String> allSourceExtensions = buildExtensionsList(true);
 
-    private static final List<String> allExtensions = buildExtensionsList();
+    private static final Set<String> allTranslationExtensions = buildExtensionsList(false);
 
-    private static List<String> buildExtensionsList() {
-        List<String> allExtensions = new ArrayList<String>();
+    private static Map<String, String> buildPotMap() {
+        Map<String, String> potMap = new HashMap<>();
+        potMap.put("pot", "po");
+        return unmodifiableMap(potMap);
+    }
+
+    private static Set<String> buildExtensionsList(boolean source) {
+        Set<String> allExtensions = new HashSet<>();
         for (DocumentType type : DocumentType.values()) {
-            allExtensions.addAll(type.extensions);
+            if(source) {
+                allExtensions.addAll(type.getSourceExtensions());
+            } else {
+                allExtensions.addAll(type.getTranslationExtensions());
+            }
+        }
+        return unmodifiableSet(allExtensions);
+    }
+
+    public static @Nullable DocumentType getByName(String name) {
+        try {
+            return DocumentType.valueOf(name.toUpperCase());
+        } catch (Exception e) {
+            return null;
         }
-        return unmodifiableList(allExtensions);
     }
 
     /**
-     * @return a read-only list of file extensions for known file types
+     * @return a read-only list of source file extensions for known file types
      */
-    public static List<String> getAllExtensions() {
-        return allExtensions;
+    public static Set<String> getAllSourceExtensions() {
+        return allSourceExtensions;
+    }
+
+    /**
+     * @return a read-only list of translation file extensions for known file types
+     */
+    public static Set<String> getAllTranslationExtensions() {
+        return allTranslationExtensions;
+    }
+
+    public static Set<DocumentType> fromSourceExtension(String sourceExtension) {
+        Set<DocumentType> documentTypes = new HashSet<>();
+        for (DocumentType type : DocumentType.values()) {
+            if (type.extensions.keySet().contains(sourceExtension)) {
+                documentTypes.add(type);
+            }
+        }
+        return documentTypes;
     }
 
-    // FIXME damason: rename typeFor to fromString
-    public static DocumentType typeFor(String extension) {
+    public static Set<DocumentType> fromTranslationExtension(
+            String translationExtension) {
+        Set<DocumentType> documentTypes = new HashSet<>();
         for (DocumentType type : DocumentType.values()) {
-            if (type.extensions.contains(extension)) {
-                return type;
+            if (type.extensions.values().contains(translationExtension)) {
```

Comment:
The code in this method is almost identical to `fromSourceExtension`. The only different bit is `.values()` instead of `.keySet()`.  We could use an inline function from `Map<String, String> -> List<String>` to inject that behaviour into a common piece of code. Probably not worth it until we have switched to Java 8 though.